File: scripts/convert-icons.js
/**
 * 这个脚本用于将 SVG 图标转换为微信小程序 tabBar 所需的 PNG 格式
 * 
 * 使用方法：
 * 1. 安装 sharp: npm install sharp
 * 2. 运行: node scripts/convert-icons.js
 * 
 * 或者手动使用在线工具转换：
 * - https://cloudconvert.com/svg-to-png
 * - https://svgtopng.com/
 * 
 * 微信小程序 tabBar 图标要求：
 * - 格式: PNG/JPG/JPEG
 * - 尺寸: 81x81 像素（建议）
 * - 大小: 不超过 40KB
 */

const fs = require('fs');
const path = require('path');

// 检查是否安装了 sharp
let sharp;
try {
    sharp = require('sharp');
} catch (e) {
    console.log('请先安装 sharp: npm install sharp');
    console.log('或者手动使用在线工具将以下 SVG 文件转换为 PNG:');
    console.log('');

    const svgDir = path.join(__dirname, '../assets/icons');
    const svgFiles = fs.readdirSync(svgDir).filter(f => f.endsWith('.svg'));

    svgFiles.forEach(file => {
        console.log(`  - ${file}`);
    });

    console.log('');
    console.log('转换后的 PNG 文件放在 assets/icons/tab/ 目录下');
    process.exit(1);
}

const iconNames = ['home', 'chat', 'notes', 'quiz', 'profile'];
const srcDir = path.join(__dirname, '../assets/icons');
const destDir = path.join(__dirname, '../assets/icons/tab');

async function convertIcon(name) {
    const normalSvg = path.join(srcDir, `${name}.svg`);
    const activeSvg = path.join(srcDir, `${name}-active.svg`);

    const normalPng = path.join(destDir, `${name}.png`);
    const activePng = path.join(destDir, `${name}-active.png`);

    if (fs.existsSync(normalSvg)) {
        await sharp(normalSvg)
            .resize(81, 81)
            .png()
            .toFile(normalPng);
        console.log(`✓ 转换: ${name}.svg -> ${name}.png`);
    }

    if (fs.existsSync(activeSvg)) {
        await sharp(activeSvg)
            .resize(81, 81)
            .png()
            .toFile(activePng);
        console.log(`✓ 转换: ${name}-active.svg -> ${name}-active.png`);
    }
}

async function main() {
    // 确保目标目录存在
    if (!fs.existsSync(destDir)) {
        fs.mkdirSync(destDir, { recursive: true });
    }

    console.log('开始转换 SVG 图标为 PNG...\n');

    for (const name of iconNames) {
        await convertIcon(name);
    }

    console.log('\n转换完成！');
}

main().catch(console.error);

File: scripts/convert_icons.py
#!/usr/bin/env python3
"""
将 SVG 图标转换为 PNG 格式用于微信小程序 tabBar
"""
import os
import cairosvg

# 路径配置
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
SVG_DIR = os.path.join(BASE_DIR, 'assets', 'icons')
PNG_DIR = os.path.join(BASE_DIR, 'assets', 'icons', 'tab')

# 需要转换的图标
ICONS = ['home', 'chat', 'notes', 'quiz', 'profile']

# 输出尺寸 (微信小程序建议 81x81)
SIZE = 81

def convert_svg_to_png(svg_path, png_path, size=SIZE):
    """将 SVG 转换为 PNG"""
    try:
        cairosvg.svg2png(
            url=svg_path,
            write_to=png_path,
            output_width=size,
            output_height=size
        )
        print(f"✓ 转换成功: {os.path.basename(png_path)}")
        return True
    except Exception as e:
        print(f"✗ 转换失败 {os.path.basename(svg_path)}: {e}")
        return False

def main():
    # 确保输出目录存在
    os.makedirs(PNG_DIR, exist_ok=True)
    
    print("开始转换 SVG 图标为 PNG...\n")
    
    success_count = 0
    total_count = 0
    
    for icon in ICONS:
        # 普通状态
        svg_normal = os.path.join(SVG_DIR, f'{icon}.svg')
        png_normal = os.path.join(PNG_DIR, f'{icon}.png')
        
        if os.path.exists(svg_normal):
            total_count += 1
            if convert_svg_to_png(svg_normal, png_normal):
                success_count += 1
        
        # 激活状态
        svg_active = os.path.join(SVG_DIR, f'{icon}-active.svg')
        png_active = os.path.join(PNG_DIR, f'{icon}-active.png')
        
        if os.path.exists(svg_active):
            total_count += 1
            if convert_svg_to_png(svg_active, png_active):
                success_count += 1
    
    print(f"\n转换完成: {success_count}/{total_count} 个图标")
    print(f"输出目录: {PNG_DIR}")

if __name__ == '__main__':
    main()
